components/hab/src/cli.rs

```diff
@@ -264,6 +264,46 @@ pub fn get() -> App<'static, 'static> {
                     (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
                 )
             )
+            (@subcommand secret =>
+                (about: "Commands related to secret management")
+                (@setting ArgRequiredElseHelp)
+                (@subcommand generate => 
+                    (about: "Create and upload a secret for your origin")
+                    (@arg KEY_NAME: +required +takes_value
+                        "The name of the variable key to be injected into the studio.")
+                    (@arg SECRET: +required +takes_value
+                        "The contents of the variable to be injected into the studio.")
+                    (@arg BLDR_URL: -u --url +takes_value {valid_url}
+                        "Specify an alternate Builder endpoint. If not specified, the value will \
+                         be taken from the HAB_BLDR_URL environment variable if defined. (default: \
+                         https://bldr.habitat.sh)")
+                    (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
+                    (@arg ORIGIN: +takes_value
+                        "The origin for which the secret will be generated. Default is from 'HAB_ORIGIN' or cli.toml")
+                )
+                (@subcommand delete => 
+                    (about: "Delete a secret for your origin")
+                    (@arg KEY_NAME: +required +takes_value
+                        "The name of the variable key to be injected into the studio.")
+                    (@arg BLDR_URL: -u --url +takes_value {valid_url}
+                        "Specify an alternate Builder endpoint. If not specified, the value will \
+                         be taken from the HAB_BLDR_URL environment variable if defined. (default: \
+                         https://bldr.habitat.sh)")
+                    (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
+                    (@arg ORIGIN: +takes_value
```

Comment:
I wonder if it might be a bit more usable to make `ORIGIN` an option (requiring `-o, --origin`)
```
hab origin secret delete foo bar
```
is less obvious to me than
```
hab origin secret delete --origin bar foo
```
And generally, I think making the purpose of arguments that will often be omitted explicit reduces cognitive load. This goes for all the `hab origin secret` subcommands

Reply:
I agree with the idea of having origin be a flag but in an effort to remain consistent with the existing UX I left it as a positional argument. I think a larger refactor is necessary to make origin a flag everywhere to keep everything consistent

Reply:
I looked at the rest of the UX when there are multiple positional parameters, we *do* make origin an option:
```
    hab bldr job promote [FLAGS] [OPTIONS] <GROUP_ID> <CHANNEL>
…
OPTIONS:
…
    -o, --origin <ORIGIN>      Limit the promotable packages to the specified origin
```
When the origin is the *only* possible positional argument, it's fine, but in this case I think it's the UX consistent behavior to make it an option requiring a flag.

Reply:
My .02 is that I also prefer the clarity of using the -o or --origin flag... that also helps with not having concern your self with the order of the arguments - its explicit that the thing following the flag is in fact the origin name, in other words.